_stbt/core.py

```diff
@@ -1620,10 +1636,14 @@ def teardown(self):
             source = None
         if not self.novideo:
             debug("teardown: Sending eos")
-            self.appsrc.emit("end-of-stream")
-            self.mainloop_thread.join(10)
-            debug("teardown: Exiting (GLib mainloop %s)" % (
-                "is still alive!" if self.mainloop_thread.isAlive() else "ok"))
+            if self.appsrc.emit("end-of-stream") == Gst.FlowReturn.OK:
+                if not self.received_eos.wait(10):
+                    debug("Timeout waiting for sink EOS")
+            else:
+                debug("Sending EOS to sink pipeline failed")
+            self.sink_pipeline.set_state(Gst.State.NULL)
```

Comment:
Good catch with the set_state(NULL). I wonder if this will fix some very intermittent segfaults I've seen.

When we did a similar fix for the _source_ pipeline (5d258bf) it fixed some intermittent segfaults during teardown.